### PR TITLE
Fix placement nil dereference when reading stream

### PR DESCRIPTION
**Pull request by mcandeia**
Signed-off-by: Marcos Candeia <marrcooos@gmail.com>

# Description

I've removed the nil assignment for the clientStream when closing the stream that was causing a nil dereference. This should not be a problem in terms of garbage collection, given the `closeStream` calling frequency. Also, calling CloseSend and Close twice does not panic, so it will not be a problema as well.

## Issue reference

Please reference the issue this PR will close: #5575

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by mcandeia** [comment]
please review @yaron2 @artursouza ,

Another interesting fact is that I think we are not handling the stream close very well. According to the [documentation](https://github.com/grpc/grpc-go/blob/be1fb4f27549f736b9b4ec26104c7c6b29845ad0/stream.go#L106) we should cancel context and recv until receiving an EOF or any error from header.

1) We are not calling context cancel, actually we are using a [background context](https://github.com/dapr/dapr/blob/master/pkg/actors/internal/placement.go#L372) 
2) We are not reading the stream until it ends

It might lead in memory leaks. I'll create an issue for further investigation soon, but as we are going to deploy this change by the EOD or tomorrow morning, I'd rather prefer to do it as a separate issue/task.

File: pkg/actors/internal/placement.go [hunk 1]
```diff
@@ -181,11 +181,12 @@ func (p *ActorPlacement) Start() {
 			p.clientLock.RLock()
 			clientStream := p.clientStream
 			p.clientLock.RUnlock()
-			resp, err := clientStream.Recv()
+
 			if p.shutdown.Load() {
 				break
 			}
 
+			resp, err := clientStream.Recv()
```
Comment (ItalyPaleAle) [inline pkg/actors/internal/placement.go:189]: why was this moved after the `break`?

File: pkg/actors/internal/placement.go [hunk 2]
```diff
@@ -304,15 +305,8 @@ func (p *ActorPlacement) closeStream() {
 		p.clientLock.Lock()
```
Comment (ItalyPaleAle) [inline pkg/actors/internal/placement.go:305]: Another (unrealted) thing is that we should switch from "uber/atomic" to the built-in atomic package in the stdlib

**Review by ItalyPaleAle — APPROVED** [review]

**Review by ItalyPaleAle — APPROVED** [review]

File: pkg/actors/internal/placement.go (hunk 1 above)
Comment (mcandeia) [reply pkg/actors/internal/placement.go:189]: before the current change it was necessary to avoid race conditions between Stop() and this, but now there are no reasons to keep this, so I'm going to put it back there, thanks

File: pkg/actors/internal/placement.go [hunk 3]
```diff
@@ -181,11 +181,12 @@ func (p *ActorPlacement) Start() {
 			p.clientLock.RLock()
```
[the remark this answers is not in the document]
Comment (mcandeia) [reply pkg/actors/internal/placement.go]: I agree, I think it is just a convenience way of using "defer" without needing to duplicate the `.Unlock` you know? but there are alternatives for sure.

File: pkg/actors/internal/placement.go [hunk 4]
```diff
@@ -306,12 +306,10 @@ func (p *ActorPlacement) closeStream() {
 
 		if p.clientStream != nil {
 			p.clientStream.CloseSend()
-			p.clientStream = nil
 		}
 
 		if p.clientConn != nil {
 			p.clientConn.Close()
-			p.clientConn = nil
```
Comment (ItalyPaleAle) [inline pkg/actors/internal/placement.go:314]: Now that the nil-check is back.... what happens if you call Close twice on the same clientConn?

Comment (mcandeia) [reply pkg/actors/internal/placement.go:314]: Great question, I've tested it out because I had the same doubt, but there are no panics thanks to the [clientconn check](https://github.com/grpc/grpc-go/blob/master/clientconn.go#L1019-L1024)

**Review by yaron2 — APPROVED** [review]
